app/components/pages/SubmissionWizard/steps/Files/SupportingUpload.js

```diff
@@ -162,32 +163,45 @@ class SupportingUpload extends React.Component {
 
   synchronouslyUploadFiles = files => {
     const { uploadFile } = this.props
-    const iterator = fileUploadGenerator(files, uploadFile)
-    const loop = result => {
+    const { removeSingleFile } = this.props
+    const iterator = fileUploadGenerator(files, uploadFile, removeSingleFile)
+    const loop = async result => {
       if (result.done) {
         this.setState({ uploading: false })
       } else {
-        result.value.upload
-          .then(data => {
-            const updatedFile = data.data.uploadSupportingFile.files.filter(
-              file => file.filename === result.value.filename,
-            )[0]
-            this.updateFileState(
-              result.value.fileId,
-              {
-                success: true,
+        const index = this.state.files.findIndex(
+          file =>
+            (file.file.filename || file.file.name) === result.value.filename,
+        )
+
+        if (index > -1) {
+          await result.value.remove(this.state.files[index].file.id)
+          // TODO: now we have to upload the new one.
+          // console.log('now we have to upload the file again!')
+          // console.log(response)
+        } else {
```

Comment:
This may not need to be an `else`.

We can remove the file if it's a duplicate, and always upload the new file the user is trying to upload.